eclass/dist-kernel-utils.eclass

```diff
@@ -106,10 +106,21 @@ dist-kernel_install_kernel() {
 		# install the combined executable in place of kernel
 		image=${initrd}.efi
 		mv "${initrd}" "${image}" || die
-		# put an empty file in place of initrd.  installing a duplicate
-		# file would waste disk space, and removing it entirely provokes
-		# kernel-install to regenerate it via dracut.
-		> "${initrd}"
+		# We moved the generated initrd, prevent dracut from running again
```

Comment:
Currently, if we get rid of this workaround:
If layout=bls (the default):

Dracut creates uki with the name initrd (confusing), we move and rename the uki so it gets picked up by 90-copy-uki.install. If dracut is run again, it will create the uki with the name initrd again and install it to $BOOT. Now we have two identical uki's in $BOOT one named initrd and one named linux. System boots, but it is confusing and a wast of space.

If layout=uki (what you probably want if you instruct dracut to make uki's since in this layout the uki ends up on the ESP):

Install breaks because 50-dracut.install does not provide dracut with the --kernel-image argument which is required if we want to build a uki.

Skipping the dracut install plugins solves both problems (without creating a confusing empty initrd file).


After applying the upstream PR I linked, getting rid of this workaround, and ensuring in the eclass that we rename the generated uki to uki.efi. Things just work since 50-dracut.install now finds the pregenerated uki.efi and copies it to where 90-copy-uki.install looks for it.

Reply:
For the record and for clarity. The idea is that once the upstream PR with the fix https://github.com/dracutdevs/dracut/pull/2405 is merged and there are no old versions without the patch left in ::gentoo. The following patch (or similar) can be applied:
```
diff --git a/eclass/dist-kernel-utils.eclass b/eclass/dist-kernel-utils.eclass
index e371e035c856..d5b22403b5d7 100644
--- a/eclass/dist-kernel-utils.eclass
+++ b/eclass/dist-kernel-utils.eclass
@@ -104,23 +104,8 @@ dist-kernel_install_kernel() {
if [[ ${magic} == MZ ]]; then
einfo "Combined UEFI kernel+initramfs executable found"
# install the combined executable in place of kernel
-               image=${initrd}.efi
+               image=${initrd%/*}/uki.efi
mv "${initrd}" "${image}" || die
-               # We moved the generated initrd, prevent dracut from running again
-               local plugins=()
-               for file in "${EROOT}"/usr/lib/kernel/install.d/*; do
-                       if [[ ${file} != */50-dracut.install && \
-                               ${file} != */51-dracut-rescue.install && \
-                               ${file} == *.install ]]; then
-                                       plugins+=( "${file}" )
-                       fi
-               done
-               for file in "${EROOT}"/etc/kernel/install.d/*; do
-                       if [[ ${file} == *.install ]]; then
-                                       plugins+=( "${file}" )
-                       fi
-               done
-               export KERNEL_INSTALL_PLUGINS="${KERNEL_INSTALL_PLUGINS} ${plugins[@]}"
fi

ebegin "Installing the kernel via installkernel"
```

The PR also fixes regular `make install` with manually configured kernels if uefi=yes and layout=uki.

Reply:
Could I rely on you remembering to remove it whenever it's fine to remove it?

Reply:
Yes, once the PR is merged I'll keep an eye on the dracut package and once the fix has fully propagated into ::gentoo I'll get rid of this workaround.